components/server/src/workspace/messagebus-integration.ts

```diff
@@ -127,10 +129,11 @@ export class MessageBusIntegration extends AbstractMessageBusIntegration {
         return Disposable.create(() => cancellationTokenSource.cancel())
     }
 
-    listenForWorkspaceInstanceUpdates(userId: string | undefined, callback: (ctx: TraceContext, workspaceInstance: WorkspaceInstance) => void): Disposable {
+    async listenForWorkspaceInstanceUpdates(userId: string | undefined, callback: (ctx: TraceContext, workspaceInstance: WorkspaceInstance) => void): Disposable {
```

Comment:
this method cannot be async

Reply:
the same for other listen methods which return disposable object, it should happen synchronously to be registered fro the current server

Reply:
Very good point, changing it to synchronously would make us lose the ability to use `listener.topic()` which returns the topic name.

I guess this one won't make into tomorrow's release then, I need to think of another way to get the name

Reply:
you can do `listener.topic().then(increaseMessagebusTopicReads, () => { /* no-op*/ })`

Reply:
We could also try to get rid of the `async` in front of `topic` @ArthurSens ping me if you find time for this today.